fda-model/structures/Inventory.cs

```diff
@@ -95,148 +109,136 @@ public static T TryGetObj<T>(object value, T defaultValue = default)
+        createColumnHeadersForMissingColumns(ref structureInventory, _map);
+        PolygonFeatureLayer impactAreaSet = new PolygonFeatureLayer("Impact_Area_Set", _impactAreaShapefile);
+
+        float[] groundelevs = Array.Empty<float>();
+        if (_updateGroundElevsFromTerrain)
         {
-            layerColumnNames.Add(c.ColumnName);
+            groundelevs = GetGroundElevationFromTerrain(_structureInventoryShapefile, _terrainPath);
         }
 
-        foreach (Tuple<string,Type> nameTypePair in map.ColumnHeaders)
+        PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
+        for (int i = 0; i < structureInventory.FeatureCount(); i++)
         {
-            if (!layerColumnNames.Contains(nameTypePair.Item1))
+            //required parameters
+            PointM point = pointMs[i];
+            var row = structureInventory.FeatureRow(i);
+            int fid = TryGet<int>(row[_map.StructureID], -999);
+            double val_struct = TryGet<double>(row[_map.StructureValue], -999);
+            string st_damcat = TryGetObj<string>(row[_map.DamageCatagory], "NA");
+            string occtype = TryGetObj<string>(row[_map.OccupancyType], "NA");
+            //semi-required. We'll either have ff_elev given to us, or both ground elev and found_ht
+            double found_ht = TryGet<double>(row[_map.FoundationHeight], -999); //not gauranteed
+            double ground_elv;
+            if (_updateGroundElevsFromTerrain)
             {
-                layer.AddAttributeColumn(nameTypePair.Item1, nameTypePair.Item2);
+                ground_elv = groundelevs[i];
+            }
+            else
+            {
+                ground_elv = TryGet<double>(row[_map.GroundElev], -999); //not gauranteed
+            }
+            double ff_elev = TryGet<double>(row[_map.FirstFloorElev], -999); // not gauranteed
+            if (row[_map.FirstFloorElev] == DBNull.Value)
+            {
+                ff_elev = ground_elv + found_ht;
             }
+            //optional parameters
+            double val_cont = TryGet<double>(row[_map.ContentValue], 0);
+            double val_vehic = TryGet<double>(row[_map.VehicalValue], 0);
+            double val_other = TryGet<double>(row[_map.OtherValue], 0);
+            string cbfips = TryGetObj<string>(row[_map.CBFips], "NA");
+            double beginningDamage = TryGet<double>(row[_map.BeginningDamageDepth], 0);
+            int numStructures = TryGet<int>(row[_map.NumberOfStructures], 1);
+            int yearInService = TryGet<int>(row[_map.YearInConstruction], -999);
+            //TODO: handle number 
+            int impactAreaID = GetImpactAreaFID(point);
+            Structures.Add(new Structure(fid, point, ff_elev, val_struct, st_damcat, occtype, impactAreaID, val_cont, val_vehic, val_other, cbfips, beginningDamage, ground_elv, found_ht, yearInService, numStructures));
```

Comment:
at some point we should consider some validation here, make sure that we have values for all the required attributes and send messages when we run into structures that don't have values for those attributes